stl/inc/xstring

```diff
@@ -2905,6 +3009,12 @@ private:
             const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
             auto _Right_al_non_const = _Right_al;
             const auto _New_ptr      = _Right_al_non_const.allocate(_New_capacity); // throws
+
+            // if (_STD is_constant_evaluated()) { // Copying into uninitialized memory is technically UB, so fill it

```

Comment:
I think this is not supposed to be commented?

I assume we need more test coverage if we were not able to trigger this code path

Reply:
Yup -- this was unintentional (commenting out changes when debugging tests).  This was I'll push a fix restoring this. 

I haven't reviewed tests yet, but we may need to expand coverage, yes.  Though with these new changes, we're now ICEing in `VSO-0102478_moving_allocators` where we weren't before so who knows -- maybe this codepath is being triggered.  I'll have to do some more investigation.